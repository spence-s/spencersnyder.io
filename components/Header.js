import { useEffect } from 'react';
import Script from 'next/script';
import { useRouter } from 'next/router';
import Head from 'next/head';
import { NextSeo } from 'next-seo';
import * as gtag from '../config/analytics';

export const Header = ({ title, description, ogTitle }) => {
	const router = useRouter();
	useEffect(() => {
		const handleRouteChange = (url) => {
			gtag.pageview(url);
		};

		router.events.on('routeChangeComplete', handleRouteChange);
		return () => {
			router.events.off('routeChangeComplete', handleRouteChange);
		};
	}, [router.events]);

	return (
		<>
			<Script
				strategy='afterInteractive'
				src={`https://www.googletagmanager.com/gtag/js?id=${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}`}
			/>
			<Script
				id='gtag-init'
				strategy='afterInteractive'
				dangerouslySetInnerHTML={{
					__html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}', {
              page_path: window.location.pathname,
            });
          `
				}}
			/>
			<Head>
				<title>{title}</title>
				<NextSeo
					defaultTitle='spencersnyder.io'
					title={title}
					description={description}
					openGraph={{ title: ogTitle || title }}
				/>
				<link
					rel='preload'
					href='/fonts/DankMono-Regular.otf'
					as='font'
					crossOrigin=''
				/>
				<link
					rel='preload'
					href='/fonts/DankMono-Italic.otf'
					as='font'
					crossOrigin=''
				/>
			</Head>
		</>
	);
};

export default Header;
