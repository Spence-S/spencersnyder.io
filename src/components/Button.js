import * as React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledButton = styled.button`
  display: inline-block;
  min-width: 15em;
  min-height: 3em;
  margin: 0 1rem 0.3rem 0;
  box-sizing: border-box;
  text-decoration: none;
  border: 2px solid purple;
  border-radius: ${props => (props.rounded ? '.5rem' : '')};
  text-align: center;
  cursor: pointer;
  transition: all 0.3s;

  :hover {
    background-color: purple;
    color: white;
    transform: scale(1.1);
  }
`;

const Button = ({ children, ...props }) => <StyledButton {...props}>{children}</StyledButton>;

Button.propTypes = {
  children: PropTypes.node.isRequired
};

export default Button;
