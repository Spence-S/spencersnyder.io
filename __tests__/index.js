import 'react-testing-library/cleanup-after-each';
import React from 'react';
import { render } from 'react-testing-library';
import Button from '../src/components/Button';
import Header from '../src/components/header';
import 'jest-styled-components';

test('Button matches snapshot', () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Header matches snapshot', () => {
  const { container } = render(<Header />);
  expect(container.firstChild).toMatchSnapshot();
});
