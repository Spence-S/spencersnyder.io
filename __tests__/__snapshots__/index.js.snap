// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
.c0 {
  display: inline-block;
  min-width: 15em;
  min-height: 3em;
  margin: 0 1rem 0.3rem 0;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 2px solid purple;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0:hover {
  background-color: purple;
  color: white;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<button
  class="c0"
/>
`;

exports[`Header matches snapshot 1`] = `
.c0 {
  background: black;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2em;
  margin: 0;
  padding: 0;
  cursor: pointer;
}

.c4 {
  display: block;
  color: white;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  color: white;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
}

.c2:hover {
  border-bottom: 5px solid purple;
}

.c2:hover .c3 {
  color: purple;
}

@media (max-width:700px) {
  .c0 {
    position: fixed;
    bottom: 0;
  }
}

@media (max-width:700px) {
  .c1 {
    height: 2.5em;
  }
}

@media (max-width:700px) {
  .c5 {
    visibility: hidden;
  }
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3 c4"
        href="/"
      >
        <span
          class="c5"
        >
           
          Home
           
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
        >
          <path
            d="M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z"
          />
        </svg>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3 c4"
        href="about"
      >
        <span
          class="c5"
        >
           
          About
           
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 192 512"
          width="1em"
        >
          <path
            d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
          />
        </svg>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3 c4"
        href="portfolio"
      >
        <span
          class="c5"
        >
           
          Portfolio
           
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
        >
          <path
            d="M480 416v16c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V176c0-26.51 21.49-48 48-48h16v208c0 44.112 35.888 80 80 80h336zm96-80V80c0-26.51-21.49-48-48-48H144c-26.51 0-48 21.49-48 48v256c0 26.51 21.49 48 48 48h384c26.51 0 48-21.49 48-48zM256 128c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-96 144l55.515-55.515c4.686-4.686 12.284-4.686 16.971 0L272 256l135.515-135.515c4.686-4.686 12.284-4.686 16.971 0L512 208v112H160v-48z"
          />
        </svg>
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3 c4"
        href="blog"
      >
        <span
          class="c5"
        >
           
          Blog
           
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 640 512"
          width="1em"
        >
          <path
            d="M48 0C21.53 0 0 21.53 0 48v64c0 8.84 7.16 16 16 16h80V48C96 21.53 74.47 0 48 0zm208 412.57V352h288V96c0-52.94-43.06-96-96-96H111.59C121.74 13.41 128 29.92 128 48v368c0 38.87 34.65 69.65 74.75 63.12C234.22 474 256 444.46 256 412.57zM288 384v32c0 52.93-43.06 96-96 96h336c61.86 0 112-50.14 112-112 0-8.84-7.16-16-16-16H288z"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;
